library_generation/utilities.sh

```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
 
 set -xeo pipefail
+utilities_script_dir=$(echo "${BASH_SOURCE}" | rev | cut -d/ -f2- | rev)
```

Comment:
Could you add a comment to explain what this line does?

Reply:
@JoeWang1127 I forgot we are calling this script from wherever, leaving our `output` folder at mercy of the user's CWD instead of forcing it to be in `library_generation`. I reverted this change but still left the utility function now relying on `$(pwd)`